tests/test_runner.py

```diff
@@ -0,0 +1,149 @@
+import os
+import pytest
+from collections import namedtuple
+import pandas as pd
+
+from src.runner import Runner
+
+TEST_SCHEMA_SUFFIX = os.environ.get("TEST_SCHEMA_SUFFIX", None)
+
+test_schema_expected = "test_data_expected"
+test_schema_actual = "test_data_actual"
+if TEST_SCHEMA_SUFFIX:
+    test_schema_actual = f"{test_schema_actual}_{TEST_SCHEMA_SUFFIX}"
+
+test_data_query = """
+    select * from :dataset_name :filter_clause
+"""
+TestDataset = namedtuple(
+    "TestDataset",
+    [
+        "table_name",
+        "filter_clause",
+        "expected_row_count",
+    ],
+)
+# TODO add all output table names to these test datasets
+test_datasets = [
+    TestDataset(
+        table_name="dcp_projects",
+        filter_clause="""
+            where dcp_name in (
+                'P2016K0159', '2023K0228', 'P2005K0122', '2021M0260'
+                )
+            """,
+        expected_row_count=4,
+    ),
+    TestDataset(
+        table_name="dcp_projectbbls",
+        filter_clause="""
+            where SUBSTRING(dcp_name, 0,10) in (
+                'P2016K0159', '2023K0228', 'P2005K0122', '2021M0260'
+                )
+            """,
+        expected_row_count=2210,
+    ),
+]
+
+
+@pytest.mark.integration()
+@pytest.mark.parametrize("test_dataset", test_datasets)
+def test_validate_expected_test_data(test_dataset):
+    runner = Runner(name=test_dataset.table_name, schema=test_schema_expected)
+    test_data_expected = runner.pg.execute_select_query(
+        base_query=test_data_query,
+        parameters={
+            "dataset_name": test_dataset.table_name,
+            "filter_clause": test_dataset.filter_clause,
+        },
+    )
+    assert len(test_data_expected) == test_dataset.expected_row_count
+
+
+@pytest.mark.integration()
+@pytest.mark.parametrize("test_dataset", test_datasets)
+def test_runner_download(test_dataset):
+    runner = Runner(
+        name=test_dataset.table_name,
+        schema=test_schema_actual,
+    )
+    runner.download()
+    # TODO assert things
```

Comment:
Still have a todo here, not sure if you want in scope of this PR?

Reply:
Multiple below as well

Reply:
yea those are our of scope for this one. I had started to do a little bit of it but decided to focus on "does it run"

maybe there's a better [codetag](https://peps.python.org/pep-0350/#mnemonics) than `TODO` for something like this? maybe `FIXME`?

Reply:
Nah TODO is to the point

Reply:
... why do they call them mnemonics?

Reply:
lol great question